### PR TITLE
[IMP] analytic,*: Add business domain choices

### PR DESCRIPTION
**Pull request by gawa-odoo**
Add choices for the business domain on applicability rules, so rules are more precise

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by vin-odoo — APPROVED** [review]
Lgtm

File: addons/account/views/account_move_views.xml [hunk 1]
```diff
@@ -889,7 +889,7 @@
                                                groups="analytic.group_analytic_accounting"
                                                optional="show"
                                                options="{'product_field': 'product_id', 'account_field': 'account_id'}"
-                                               business_domain_compute="parent.move_type in ['out_invoice', 'out_refund', 'out_receipt'] and 'sale' or parent.move_type in ['in_invoice', 'in_refund', 'in_receipt'] and 'purchase' or 'general'"/>
+                                               business_domain_compute="parent.move_type in ['out_invoice', 'out_refund', 'in_receipt'] and 'sale' or parent.move_type in ['in_invoice', 'in_refund', 'out_receipt'] and 'purchase' or 'general'"/>
```
Comment (william-andre) [inline addons/account/views/account_move_views.xml:892]: What? This doesn't make any sense

File: addons/account/views/account_reconcile_model_views.xml [hunk 2]
```diff
@@ -24,11 +24,11 @@
...
-                            <!-- TODO widget="analytic_distribution" -->
                             <field name="analytic_distribution_stored_char" invisible="1"/>
-                            <field name="analytic_distribution" invisible="1" groups="analytic.group_analytic_accounting"
-                                                       options="{'product_field': 'product_id', 'account_field': 'account_id'}"
-                                                       business_domain_compute="parent.move_type in ['out_invoice', 'out_refund', 'out_receipt'] and 'sale' or parent.move_type in ['in_invoice', 'in_refund', 'in_receipt'] and 'purchase' or 'general'"/>
+                            <field name="analytic_distribution" widget="analytic_distribution"
+                                   groups="analytic.group_analytic_accounting"
+                                   options="{'account_field': 'account_id'}"
+                                   business_domain_compute="parent.move_type in ['out_invoice', 'out_refund', 'out_receipt'] and 'invoice' or parent.move_type in ['in_invoice', 'in_refund', 'in_receipt'] and 'bill' or 'general'"/>
```
Comment (william-andre) [inline addons/account/views/account_reconcile_model_views.xml:31]: What is the move type on reconcile models?
And what is the parent on the line's form view?

Comment (gawa-odoo) [reply addons/account/views/account_reconcile_model_views.xml:31]: idem

Comment (william-andre) [reply addons/account/views/account_reconcile_model_views.xml:31]: How is this one even working? :sweat_smile:

Comment (gawa-odoo) [reply addons/account/views/account_reconcile_model_views.xml]: It wasn't before, it was in invisible. I copy-pasted too fast for this field. Fixed it

**Pull request closed without merging**